vllm/model_executor/layers/fused_moe/layer.py

```diff
@@ -302,8 +337,26 @@ def __init__(
         self.topk_group = topk_group
         self.custom_routing_function = custom_routing_function
         if is_hpu:
-            from vllm_hpu_extension.ops import DynamicFusedMOE
-            self.hpu_fused_moe = DynamicFusedMOE(self.num_experts)
+            ep_shift = self.ep_rank * self.num_experts
+            from vllm_hpu_extension.ops import (VllmMixtureOfExpertsOp,
+                                                VllmMixtureOfExpertsOpFP8)
+            moe_n_slice = 1 if self.num_experts <= 64 \
+                else self.num_experts // 64
+            num_expert_per_group = self.num_experts // moe_n_slice
+            experts_min, experts_max = ep_shift, self.num_experts + ep_shift - 1
+            if quant_config is not None:
```

Comment:
@yiliu30 , I removed VLLM_REQUANT_FP8_INC check and replaced with if this func is fp8 or unquantized